cli/flox/src/commands/edit.rs

```diff
@@ -130,18 +195,21 @@ impl Edit {
     // instead of just environment is a pain
     async fn edit_manifest(
         flox: &Flox,
-        environment: &mut Box<dyn Environment>,
+        environment: &mut dyn Environment,
         active_environment: UninitializedEnvironment,
         description: String,
         contents: Option<String>,
     ) -> Result<()> {
         match maybe_migrate_environment_to_v1(flox, environment, &description).await {
             Ok(_) => (),
             e @ Err(MigrationError::MigrationCancelled) => e?,
+
+            e @ Err(MigrationError::ConfirmedUpgradeFailed(
+                EnvironmentError::ManagedEnvironment(ManagedEnvironmentError::CheckoutOutOfSync),
+            )) => e?,
```

Comment:
question nonblocking: is this reachable? Since there's a `bail` for `has_local_changes` above

Reply:
probably not anymore no, the bail above was added later, becuase otherwise we'd get an editor _and then_ see that the manifest was out of datae which was a bit late..